### PR TITLE
GridV2 sample graders (regular feedback, rich feedback, and networking)

### PR DESCRIPTION
**Pull request by sashi-coursera**

### CODE REVIEW

File: custom-graders/v2/DemoJavaGrader/executeGrader.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+#    cd /grader
+# else
+#    cd ./grader
+# fi
+
+# Use the parsed partId to know which part is being graded in the current run.
+
+if [ $local != "1" ]
+then
+	GRADER_DIRECTORY=FactoringGrader
```
Comment (fng-coursera) [inline custom-graders/v2/DemoJavaGrader/executeGrader.sh]: GRADER_DIRECTORY=/grader/FactoringGrader

File: custom-graders/v2/DemoJavaGrader/executeGrader.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+else
+	GRADER_DIRECTORY=./FactoringGrader
```
Comment (fng-coursera) [inline custom-graders/v2/DemoJavaGrader/executeGrader.sh]: `GRADER_DIRECTORY=./grader/FactoringGrader`

File: custom-graders/v2/DemoJavaGrader/executeGrader.sh [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+# Check if the learner's program ran successfully
+if [ ! $? -eq 0 ]; then
+	echo "{ \"fractionalScore\": 0.0, \"feedback\":\"Your submission produced runtime errors (GrIDV2 stdout)\" }"
+	echo "{ \"fractionalScore\": 0.0, \"feedback\":\"Your submission produced runtime errors (GrIDV2 feedback)\" }" > /shared/feedback.json
+	exit 0
+fi
+
+# Compile Grader.java
+javac Grader.java
```
Comment (fng-coursera) [inline custom-graders/v2/DemoJavaGrader/executeGrader.sh:39]: `javac -d . /grader/Grader.java`

File: custom-graders/v2/DemoJavaGrader/Grader.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+		System.out.println("Environment variables: \n");
+        	for (String envName : env.keySet()) {
+            		System.out.format("%s=%s%n",
+                              envName,
+                              env.get(envName));
+        	}
+
+		boolean local = env.get("LOCAL") != null;
+
+		String feedbackfile = "/shared/feedback.json";
```
Comment (sashi-coursera) [inline custom-graders/v2/DemoJavaGrader/Grader.java]: Feedback now goes to : "/shared/feedback.json"

File: custom-graders/v2/DemoJavaGrader/executeGrader.sh [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+# Use the parsed partId to know which part is being graded in the current run.
+
+GRADER_DIRECTORY=FactoringGrader
+SUBMISSION_CLASS=Factoring
+
+
+# Compile the learner's program in the current directory. We can safely assume that there
+# would be a single submission file in this directory.
+javac -d . /shared/submission/$filename
```
Comment (sashi-coursera) [inline custom-graders/v2/DemoJavaGrader/executeGrader.sh]: use of environment variable.

**Pull request merged**